--- DAP_config.h.orig
+++ DAP_config.h
@@ -135,10 +135,6 @@
 #define PIN_SWDIO_TMS_PORT      1
 #define PIN_SWDIO_TMS_BIT       20
 
-// SWDIO Output Enable Pin      PIO1_21
-#define PIN_SWDIO_OE_PORT       1
-#define PIN_SWDIO_OE_BIT        21
-
 // TDI Pin                      PIO1_22
 #define PIN_TDI_PORT            1
 #define PIN_TDI_BIT             22
@@ -155,11 +151,6 @@
 #define PIN_nRESET_PORT         1
 #define PIN_nRESET_BIT          26
 
-// nRESET Output Enable Pin     PIO1_27
-#define PIN_nRESET_OE_PORT      1
-#define PIN_nRESET_OE_BIT       27
-
-
 // Debug Unit LEDs
 
 // Connected LED                PIO0_23
@@ -220,8 +211,16 @@
  - TDO to input mode.
 */ 
 static __inline void PORT_JTAG_SETUP (void) {
-  LPC_GPIO->MASK[PIN_SWDIO_TMS_PORT] = 0;
-  LPC_GPIO->MASK[PIN_TDI_PORT] = ~(1 << PIN_TDI_BIT);
+	LPC_GPIO->MASK[PIN_SWDIO_TMS_PORT] = 0;
+	LPC_GPIO->MASK[PIN_TDI_PORT] = ~(1 << PIN_TDI_BIT);
+	
+	LPC_GPIO->SET[PIN_SWCLK_TCK_PORT] = (1 << PIN_SWCLK_TCK_BIT);
+	LPC_GPIO->SET[PIN_SWDIO_TMS_PORT] = (1 << PIN_SWDIO_TMS_BIT);
+
+	LPC_GPIO->DIR[PIN_SWCLK_TCK_PORT] |= (1 << PIN_SWCLK_TCK_BIT);
+	LPC_GPIO->DIR[PIN_SWDIO_TMS_PORT] |= (1 << PIN_SWDIO_TMS_BIT);
+
+	LPC_GPIO->DIR[PIN_nRESET_PORT]    &= ~(1 << PIN_nRESET_BIT);
 }
  
 /** Setup SWD I/O pins: SWCLK, SWDIO, and nRESET.
@@ -230,8 +229,13 @@
  - TDI, TMS, nTRST to HighZ mode (pins are unused in SWD mode).
 */ 
 static __inline void PORT_SWD_SETUP (void) {
-  LPC_GPIO->MASK[PIN_TDI_PORT] = 0;
-  LPC_GPIO->MASK[PIN_SWDIO_TMS_PORT] = ~(1 << PIN_SWDIO_TMS_BIT);
+	LPC_GPIO->MASK[PIN_TDI_PORT] = 0;
+	LPC_GPIO->MASK[PIN_SWDIO_TMS_PORT] = ~(1 << PIN_SWDIO_TMS_BIT);
+	LPC_GPIO->SET[PIN_SWCLK_TCK_PORT] = (1 << PIN_SWCLK_TCK_BIT);
+	LPC_GPIO->SET[PIN_SWDIO_TMS_PORT] = (1 << PIN_SWDIO_TMS_BIT);
+	LPC_GPIO->DIR[PIN_SWCLK_TCK_PORT] |= (1 << PIN_SWCLK_TCK_BIT);
+	LPC_GPIO->DIR[PIN_SWDIO_TMS_PORT] |= (1 << PIN_SWDIO_TMS_BIT);
+	LPC_GPIO->DIR[PIN_nRESET_PORT]    &= ~(1 << PIN_nRESET_BIT);
 }
 
 /** Disable JTAG/SWD I/O Pins.
@@ -241,10 +245,8 @@
 static __inline void PORT_OFF (void) {
   LPC_GPIO->SET[PIN_SWCLK_TCK_PORT]  =  (1 << PIN_SWCLK_TCK_BIT);
   LPC_GPIO->SET[PIN_SWDIO_TMS_PORT]  =  (1 << PIN_SWDIO_TMS_BIT);
-  LPC_GPIO->SET[PIN_SWDIO_OE_PORT]   =  (1 << PIN_SWDIO_OE_BIT);
   LPC_GPIO->SET[PIN_TDI_PORT]        =  (1 << PIN_TDI_BIT);
   LPC_GPIO->DIR[PIN_nRESET_PORT]    &= ~(1 << PIN_nRESET_BIT);
-  LPC_GPIO->CLR[PIN_nRESET_OE_PORT]  =  (1 << PIN_nRESET_OE_BIT);
 }
 
 
@@ -314,7 +316,7 @@
 called prior \ref PIN_SWDIO_OUT function calls.
 */
 static __forceinline void     PIN_SWDIO_OUT_ENABLE  (void) {
-  LPC_GPIO->SET[PIN_SWDIO_OE_PORT] = 1 << PIN_SWDIO_OE_BIT;
+	LPC_GPIO->DIR[PIN_SWDIO_TMS_PORT] |= (1<<PIN_SWDIO_TMS_BIT);
 }
 
 /** SWDIO I/O pin: Switch to Input mode (used in SWD mode only).
@@ -322,7 +324,7 @@
 called prior \ref PIN_SWDIO_IN function calls.
 */
 static __forceinline void     PIN_SWDIO_OUT_DISABLE (void) {
-  LPC_GPIO->CLR[PIN_SWDIO_OE_PORT] = 1 << PIN_SWDIO_OE_BIT;
+	LPC_GPIO->DIR[PIN_SWDIO_TMS_PORT] &= ~(1<<PIN_SWDIO_TMS_BIT);
 }
 
 
@@ -388,10 +390,9 @@
 static __forceinline void     PIN_nRESET_OUT (uint32_t bit) {
   if (bit) {
     LPC_GPIO->DIR[PIN_nRESET_PORT]    &= ~(1 << PIN_nRESET_BIT);
-    LPC_GPIO->CLR[PIN_nRESET_OE_PORT]  =  (1 << PIN_nRESET_OE_BIT);
   } else {
-    LPC_GPIO->SET[PIN_nRESET_OE_PORT]  =  (1 << PIN_nRESET_OE_BIT);
-    LPC_GPIO->DIR[PIN_nRESET_PORT]    |=  (1 << PIN_nRESET_BIT);
+	LPC_GPIO->DIR[PIN_nRESET_PORT]    |=  (1 << PIN_nRESET_BIT);
+	LPC_GPIO->CLR[PIN_nRESET_PORT] = 1<<PIN_nRESET_BIT;
   }
 }
 
@@ -427,7 +428,7 @@
 */
 static __inline void LED_RUNNING_OUT (uint32_t bit) {
   //;             // Not available
-	LPC_GPIO->B[32*LED_CONNECTED_PORT + LED_CONNECTED_BIT] = bit;
+	LPC_GPIO->B[32*LEDRUNNING_CONNECTED_PORT + LEDRUNNING_CONNECTED_BIT] = bit;
 }
 
 ///@}
@@ -453,33 +454,26 @@
 static __inline void DAP_SETUP (void) {
 
   /* Configure I/O pins																				 */	
-	LPC_IOCON->PIO1_19 = FUNC_0 | PULL_UP_ENABLED;  /* SWCLK/TCK */
-	LPC_IOCON->PIO1_20 = FUNC_0 | PULL_UP_ENABLED;  /* SWDIO/TMS */
-	LPC_IOCON->PIO1_21 = FUNC_0 | PULL_UP_ENABLED;  /* SWDIO_OE  */
-	LPC_IOCON->PIO1_22 = FUNC_0 | PULL_UP_ENABLED;  /* TDI:      */
-	LPC_IOCON->PIO1_25 = FUNC_0 | PULL_UP_ENABLED;  /* TDO:      */
-	LPC_IOCON->PIO1_26 = FUNC_0 | OPENDRAIN;        /* nRESET:   */
-	LPC_IOCON->PIO1_27 = FUNC_0 | PULL_UP_ENABLED;  /* nRESET_OE */
-	LPC_IOCON->PIO0_23 = FUNC_0 | PULL_UP_ENABLED;  /* LED:      */
-	LPC_IOCON->PIO0_22 = FUNC_0 | PULL_UP_ENABLED;  /* LED:      */
+	LPC_IOCON->PIO1_19 = FUNC_0;  /* SWCLK/TCK */
+	LPC_IOCON->PIO1_20 = FUNC_0;  /* SWDIO/TMS */
+	LPC_IOCON->PIO1_22 = FUNC_0;  /* TDI:      */
+	LPC_IOCON->PIO1_25 = FUNC_0;  /* TDO:      */
+	LPC_IOCON->PIO1_26 = FUNC_0 | OPENDRAIN | PULL_UP_ENABLED;        /* nRESET:   */
+	LPC_IOCON->PIO0_23 = FUNC_0;  /* LED:      */
+	LPC_IOCON->PIO0_22 = FUNC_0;  /* LED:      */
 
   /* Configure: SWCLK/TCK, SWDIO/TMS, SWDIO_OE, TDI as outputs (high level)  */
   /*            TDO as input                                                 */
   /*            nRESET as input with output latch set to low level           */
-  /*            nRESET_OE as output (low level)                              */
   LPC_GPIO->SET[PIN_SWCLK_TCK_PORT]  =  (1 << PIN_SWCLK_TCK_BIT);
   LPC_GPIO->SET[PIN_SWDIO_TMS_PORT]  =  (1 << PIN_SWDIO_TMS_BIT);
-  LPC_GPIO->SET[PIN_SWDIO_OE_PORT]   =  (1 << PIN_SWDIO_OE_BIT);
   LPC_GPIO->SET[PIN_TDI_PORT]        =  (1 << PIN_TDI_BIT);
-  LPC_GPIO->CLR[PIN_nRESET_PORT]     =  (1 << PIN_nRESET_BIT);
-  LPC_GPIO->CLR[PIN_nRESET_OE_PORT]  =  (1 << PIN_nRESET_OE_BIT);
+  LPC_GPIO->SET[PIN_nRESET_PORT]     =  (1 << PIN_nRESET_BIT);
   LPC_GPIO->DIR[PIN_SWCLK_TCK_PORT] |=  (1 << PIN_SWCLK_TCK_BIT);
   LPC_GPIO->DIR[PIN_SWDIO_TMS_PORT] |=  (1 << PIN_SWDIO_TMS_BIT);
-  LPC_GPIO->DIR[PIN_SWDIO_OE_PORT]  |=  (1 << PIN_SWDIO_OE_BIT);
   LPC_GPIO->DIR[PIN_TDI_PORT]       |=  (1 << PIN_TDI_BIT);
   LPC_GPIO->DIR[PIN_TDO_PORT]       &= ~(1 << PIN_TDO_BIT);
   LPC_GPIO->DIR[PIN_nRESET_PORT]    &= ~(1 << PIN_nRESET_BIT);
-  LPC_GPIO->DIR[PIN_nRESET_OE_PORT] |=  (1 << PIN_nRESET_OE_BIT);
 
   /* Configure: LED as output (turned off) */
   LPC_GPIO->CLR[LED_CONNECTED_PORT]  =  (1 << LED_CONNECTED_BIT);
